### PR TITLE
fix: wrong redirect when skipping recommended apps

### PR DESCRIPTION
**Pull request by kesselb**
* Resolves: # 

## Summary

To access the page after setup: `index.php/core/apps/recommended`

**Before**

[Screencast from 2023-03-13 20-49-14.webm](https://user-images.githubusercontent.com/3902676/224959336-ddbc3490-50d1-450f-bc8d-21684c8ef337.webm)

**After**

[Screencast from 2023-03-13 20-48-10.webm](https://user-images.githubusercontent.com/3902676/224959365-9b75b02c-a145-4d81-9b72-9bd3708246f0.webm)

## TODO

- [x] CI

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: core/src/components/setup/RecommendedApps.vue [hunk 1]
```diff
@@ -59,7 +59,6 @@
 			<NcButton v-if="showInstallButton"
 				type="tertiary"
 				role="link"
-				href="defaultPageUrl"
```
Comment (tcitworld) [inline core/src/components/setup/RecommendedApps.vue:62]: **suggestion:** We can keep it a link, just set
```vue
:href="defaultPageUrl"
```
and remove the `@click` event (and the useless method `goTo`).

Comment (kesselb) [reply core/src/components/setup/RecommendedApps.vue:62]: Sounds good. `defaultPageUrl` is defined outside the vue component hence we need an computed property to access the variable from the template.

File: core/src/components/setup/RecommendedApps.vue [hunk 2]
```diff
@@ -129,10 +128,12 @@ export default {
 			loadingApps: true,
 			loadingAppsError: false,
 			apps: [],
-			defaultPageUrl,
 		}
 	},
 	computed: {
+		defaultPageUrl() {
+			return defaultPageUrl
+		},
```
Comment (tcitworld) [inline core/src/components/setup/RecommendedApps.vue:136]: **suggestion (non-blocking):** Data would be fine as well, since the value will not change.

Comment (kesselb) [reply core/src/components/setup/RecommendedApps.vue:136]: Thanks! 

image

This version, with defaultPageUrl as a dedicated variable, does not seem to work. 

image

It seems you need a computed property to access global variables or use loadState in data directly.

**Review by Pytal — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**